lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -632,12 +633,31 @@ protected function prepareUpdateData($entity)
             $newVal = $change[1];
 
             if ( ! isset($this->class->associationMappings[$field])) {
-                $fieldMapping = $this->class->fieldMappings[$field];
+                $class = $this->class;
+
+                // Fix for bug GH-8229 (id column from parent class renamed in child class):
+                // Get the correct class metadata
+                foreach ($class->parentClasses as $parentClassName) {
+                    $parentClass = $this->em->getClassMetadata($parentClassName);
+                    if (array_key_exists($field, $parentClass->fieldMappings)) {
+                        $class = $parentClass;
+                    }
+                }
+
+                $fieldMapping = $class->fieldMappings[$field];
                 $columnName   = $fieldMapping['columnName'];
 
                 $this->columnTypes[$columnName] = $fieldMapping['type'];
 
+                // Fix for bug GH-8229 (id column from parent class renamed in child class):
+                // The getOwningTable() method is overwritten in class JoinedSubclassPersister
```

Comment:
Should it be fixed there, then? Current approach of fixing it means people using getOwningTable standalone will still have old, buggy behaviour, right? These fixes are for some reason localized to few methods only.

Reply:
That's right. I will modify it and move this part to an own method that will always be called by getOwningTable().

Reply:
Just realized that this isn't necessary anymore, because the correction of the "inherited" property in the column metadata (added alter to fix missing foreign keys) also fixed that problem, so I wll simply remove this part. :)